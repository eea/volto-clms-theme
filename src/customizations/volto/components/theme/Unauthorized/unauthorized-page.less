div.unauthorized-page span.header-login-link {
  color: #a0b128;

  &:hover {
    color: #7c8921 !important;
  }
}
