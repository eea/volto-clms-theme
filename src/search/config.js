import { mergeConfig } from '@eeacms/search';
import { getClientProxyAddress, getSearchThumbUrl } from './utils';
import vocabs from './vocabulary';
import facets from './facets';
import views from './views';

const clmsConfig = {
  title: 'CLMS Main',
  ...views,
};

export default function install(config) {
  const envConfig = process.env.RAZZLE_ENV_CONFIG
    ? JSON.parse(process.env.RAZZLE_ENV_CONFIG)
    : clmsConfig;

  const pjson = require('@eeacms/volto-clms-theme/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.clmsSearchTechnicalLibrary = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/clmsSearchTechnicalLibrary',
    index_name: 'copernicus_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
    ...vocabs,
  };

  const { clmsSearchTechnicalLibrary } = config.searchui;

  clmsSearchTechnicalLibrary.permanentFilters.push({
    term: {
      cluster_name: 'copernicus_land',
    },
  });

  clmsSearchTechnicalLibrary.facets = facets;

  clmsSearchTechnicalLibrary.resultItemModel = {
    factory: 'ResultModel',
    urlField: 'about',
    titleField: 'title',
    metatypeField: 'objectProvides',
    descriptionField: 'description',
    tagsField: 'topic',
    issuedField: 'issued',
    getThumbnailUrl: 'getSearchThumbUrl',
    getIconUrl: 'getGlobalsearchIconUrl',
  };

  clmsSearchTechnicalLibrary.sortOptions = [
    {
      name: 'Relevance',
      value: '',
      direction: '',
    },
    {
      name: 'Newest',
      value: 'issued.date',
      direction: 'desc',
    },
    {
      name: 'Oldest',
      value: 'issued.date',
      direction: 'asc',
    },
    {
      name: 'Version ↑',
      value: 'version.keyword',
      direction: 'asc',
    },
    {
      name: 'Version ↓',
      value: 'version.keyword',
      direction: 'desc',
    },
  ];
  config.resolve.getSearchThumbUrl = getSearchThumbUrl();

  if (typeof window !== 'undefined') {
    config.searchui.clmsSearchTechnicalLibrary.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }

  return config;
}
