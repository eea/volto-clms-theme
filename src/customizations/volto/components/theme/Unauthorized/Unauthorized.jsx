/**
 * @module components/theme/Unauthorized/Unauthorized
 */
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Container } from 'semantic-ui-react';
import './unauthorized-page.less';

import { withServerErrorCode } from '@plone/volto/helpers/Utils/Utils';
import CclLoginModal from '@eeacms/volto-clms-theme/components/CclLoginModal/CclLoginModal';

/**
 * unauthorized function.
 * @function Unauthorized
 * @returns {string} Markup of the unauthorized page.
 */
const Unauthorized = (props) => {
  const error_message = useSelector((state) => state.apierror.message);
  const { location = undefined } = props;

  return (
    <Container className="view-wrapper unauthorized-page">
      <h1>
        <FormattedMessage id="Unauthorized" defaultMessage="Unauthorized" />
      </h1>
      <h3>{error_message}</h3>
      <p className="description">
        <FormattedMessage
          id="You are trying to access a protected resource, please {login} first."
          defaultMessage="You are trying to access a protected resource, please {login} first."
          values={{
            login: <CclLoginModal otherPath={location?.pathname} />,
          }}
        />
      </p>
      <p>
        <FormattedMessage
          id="If you are certain you have the correct web address but are encountering an error, please contact the {site_admin}."
          defaultMessage="If you are certain you have the correct web address but are encountering an error, please contact the {site_admin}."
          values={{
            site_admin: (
              <Link to="/contact-form">
                <FormattedMessage
                  id="Site Administration"
                  defaultMessage="Site Administration"
                />
              </Link>
            ),
          }}
        />
      </p>
      <p>
        <FormattedMessage id="Thank you." defaultMessage="Thank you." />
      </p>
    </Container>
  );
};

export default withServerErrorCode(401)(Unauthorized);
