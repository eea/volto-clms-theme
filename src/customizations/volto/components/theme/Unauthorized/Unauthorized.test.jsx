import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import loadable from '@loadable/component';
import config from '@plone/volto/registry';

import Unauthorized from './Unauthorized';

const mockStore = configureStore();

// this doesnt work => loadables does not contain fontAwesome, fontAwesomeLibrary, fontAwesomeSolid, fontAwesomeRegular
jest.mock('@plone/volto/helpers/Loadable/Loadable');
beforeAll(async () => {
  await require('@plone/volto/helpers/Loadable/Loadable').__setLoadables();
});

describe('Unauthorized', () => {
  // for some reason this does not work, it does not set the loadables in the config

  it('renders a not found component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
      apierror: {
        message: 'You are not authorized to access this resource',
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Unauthorized />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
