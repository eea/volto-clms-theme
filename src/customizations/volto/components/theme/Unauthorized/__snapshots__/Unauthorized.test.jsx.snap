// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized renders a not found component 1`] = `
<div
  className="ui container view-wrapper"
>
  <h1>
    Unauthorized
  </h1>
  <h3>
    You are not authorized to access this resource
  </h3>
  <p
    className="description"
  >
    You are trying to access a protected resource, please 
    <a
      className={null}
      href="/en/login"
      onClick={[Function]}
      target={null}
      title={null}
    >
      Register/Login
    </a>
     first.
  </p>
  <p>
    If you are certain you have the correct web address but are encountering an error, please contact the 
    <a
      href="/contact-form"
      onClick={[Function]}
    >
      Site Administration
    </a>
    .
  </p>
  <p>
    Thank you.
  </p>
</div>
`;
